### ISSUE TITLE
Make the different implementations more accessible via a web interface

### ISSUE THREAD

**Issue by johanneskoester** [GoekeLab/bioinformatics-workflows#27]
We should create a github pages based view on this repo which allows to switch between the description of the example analysis and the different implementations from the different systems in a visually pleasant way.

I personally don't have the time at the moment, and @jonathangoeke already mentioned that while it would be very interesting for them, his team does not have the experience necessary for that. Hence, we thought it make sense to ask for help from the community via this issue. Is there anybody interested in giving it a try?

**Comment by radifar** [comment]
I'm interested, but I can probably start in mid November. I have experience with Jekyll. What did you usually use for GitHub page?

Edit:
I think [jekyll-doc-theme](https://github.com/aksakalli/jekyll-doc-theme) would be nice for this, web examples using this theme would be [Open Graph Benchmark](https://ogb.stanford.edu/) and [Context Mapper](https://contextmapper.org/).

There are [many static site generator](https://github.com/collections/static-site-generators) that we can use for GitHub Pages, some of the popular option are Jekyll, Hugo, and Pelican.

**Comment by jonathangoeke** [comment]
Thanks for the suggestion @johanneskoester I very much like the idea! Do you have any example github page that you have in mind for this? Something along these lines here might be great: https://github.com/clustergarage/jekyll-code-tabs not sure if that is feasible though. A more standard template like this here might be easier to implement and could also provide an easy comparison view (https://contextmapper.org/docs/home/, thanks @radifar )

**Comment by johanneskoester** [comment]
@radifar great to hear that! From the technology side, I have no preference. Personally I have used [zola](https://www.getzola.org/) and was very happy with it (it's rust, so I like it obviously :-), it has nice predefined github actions and it is easy to deploy, downside is that the theme choice might be more limited than with other frameworks), but in this case the choice is more about the preference of the developer, so if you want to do it, feel free to choose what you like.

**Comment by johanneskoester** [comment]
> I'm interested, but I can probably start in mid November. I have experience with Jekyll. What did you usually use for GitHub page?
> 
> Edit: I think [jekyll-doc-theme](https://github.com/aksakalli/jekyll-doc-theme) would be nice for this, web examples using this theme would be [Open Graph Benchmark](https://ogb.stanford.edu/) and [Context Mapper](https://contextmapper.org/).
> 
> There are [many static site generator](https://github.com/collections/static-site-generators) that we can use for GitHub Pages, some of the popular option are Jekyll, Hugo, and Pelican.

Btw. jekyll-doc-theme looks very well suited for this purpose.

**Comment by jonathangoeke** [comment]
> I'm interested, but I can probably start in mid November. 

No problem at all, I think this would be a tremendous contribution!

**Comment by radifar** [comment]
@johanneskoester I think it is possible to add `jekyll-code-tab` onto the Jekyll site, I have some experiences with adding plugin to Jekyll site.

> From the technology side, I have no preference. Personally I have used zola and was very happy with it (it's rust, so I like it obviously :-)

Jekyll is written with Ruby, but that's no problem. The thing is we can write the page easily with Markdown. Like how we write `README.md` in GitHub. I'm sure once the web deployed anyone can easily jump in.

> No problem at all, I think this would be a tremendous contribution!

I'll let you know once I can start working on it 😃

**Comment by johanneskoester** [comment]
@radifar I've seen your invite to the new bioinformatics-workflows organization. Not sure how @jonathangoeke thinks about this, but instead of a new organisation, one could also move this under github.com/compsci-commons as a repo page if you like. The reason I am suggesting this is that several workflow communities have recently created this organization exactly for such collaborative efforts between our different communities (we started with benchmarking infrastructure so far, but this page fits there equally well). I have already sent you invites, just let me know what you think.

**Comment by radifar** [comment]
Yes, sorry for not asking first. The reason I choose to create that organization is that GitHub Page would only [allow us to create the (github.io) subdomain using that username or organization name.](https://docs.github.com/en/pages/getting-started-with-github-pages/about-github-pages#types-of-github-pages-sites) This way we can use the address https://bioinformatics-workflows.github.io/. The other alternative would be to put the Jekyll site to this [repository docs](https://github.com/GoekeLab/bioinformatics-workflows/tree/master/docs). The consequences would be that the site name would be `username.github.io/projectname`, in this case means `goekelab.github.io/bioinformatics-workflows`. The third alternative would be to put it in any repository we want, and [setup custom domain](https://docs.github.com/en/pages/configuring-a-custom-domain-for-your-github-pages-site) (not free).

Anyway, I have no problem with all of the options. And I'm happy to join your invitation.

**Comment by radifar** [comment]
Oh one more thing, perhaps I should emphasize that bioinformatics-workflows is not really an organization. I just use this name for the domain sake only. I'm sorry if that brings confusion.

**Comment by jonathangoeke** [comment]
I would suggest to keep things simple and just use `goekelab.github.io/bioinformatics-workflows`? The main landing page is https://github.com/GoekeLab/bioinformatics-workflows as that is printed in the review, so this github page would be easy to find then

**Comment by radifar** [comment]
I think we all agree on this, I'll move the bioinformatics workflows repo under the docs of this repo and PR that.
